### ISSUE TITLE
Unable to change (rotate) password on stack with secrets-provider passphrase

### ISSUE THREAD

**Issue by bokjo** [pulumi/pulumi#6650]
I have a problem with rotating passphrase password

```sh 
pulumi stack change-secrets-provider passphrase
error: decrypting secret value: failed to decrypt: incorrect passphrase, please set PULUMI_CONFIG_PASSPHRASE to the correct passphrase
```

When the correct old password is set to PULUMI_CONFIG_PASSPHRASE nothing happens and I’m not prompted for a new password
```sh 
pulumi stack change-secrets-provider passphrase
Migrating old configuration and state to new secrets provider
```
According to this https://github.com/pulumi/pulumi/pull/5865 it should be available and working…

I’m trying to move it to a different backend (local storage or bucket) in order to be able to reference different stacks… I'm not able to do it since the stacks have different passwords, and I don't see an option in pulumi.StackReference to provide password I’m trying to migrate stacks from local file storage to GCS with new passwords and then later migrate again to the Pulumi SAAS…

```sh
pulumi version                                        
v2.23.2
```

## Expected behavior
Be able to change the current password of the stack with secrets-provider `passphrase`

## Current behavior
- Error is thrown when the current password is wrong
- Nothing happens when the correct password is provided (rather it is rehashed with the same password i guess )

```sh 
pulumi stack change-secrets-provider passphrase
Migrating old configuration and state to new secrets provider
```

## Steps to reproduce

1. Create a new stack with local storage or GCS bucket backend with one password
2. Save the password to `PULUMI_CONFIG_PASSPHRASE` and insert it in a file setting `PULUMI_CONFIG_PASSPHRASE_FILE` as well
3. try to change the password with a new one using `pulumi stack change-secrets-provider passphrase`

## Context (Environment)

1. Wanted to migrate stack with local file storage to GCS bucket one
2. Already have multiple stacks on GCS bucket, the idea is to reference a stack from this bucket in the stack from  point 1 (that is why migration and password change is needed, in order to be able to use `pulumi.StackReference` which doesn't support providing the password for a different stack even if they are on the same backend)
3. Prepare everything into GCS backend to later easily migrate it to the Pulumi SAAS (without data loss or redeployments)

## Affected feature
 Guess it is security!

**Comment by tusharshahrs** [comment]
Due to [6286](https://github.com/pulumi/pulumi/issues/6286)

**Comment by lyenliang** [comment]
My pulumi version is `v3.1.0`.

My `PULUMI_CONFIG_PASSPHRASE` was originally an empty string.

After running `unset PULUMI_CONFIG_PASSPHRASE`, the command `pulumi stack change-secrets-provider passphrase` is able to change the password.

```
$ unset PULUMI_CONFIG_PASSPHRASE
$ pulumi stack change-secrets-provider passphrase
Enter your new passphrase to protect config/secrets:
Re-enter your new passphrase to confirm:
Migrating old configuration and state to new secrets provider
Enter your passphrase to unlock config/secrets
    (set PULUMI_CONFIG_PASSPHRASE or PULUMI_CONFIG_PASSPHRASE_FILE to remember):
```

I'm not sure if this is an expected behavior. Because this means that anyone can change the password without knowing the old password.

**Comment by aszenz** [comment]
Just faced the same exact issue, in my case it throws an error instead of prompting for the current passphrase if the env variable `PULUMI_CONFIG_PASSPHRASE` is not defined.
```
error: constructing secrets manager of type "passphrase": unable to find either `PULUMI_CONFIG_PASSPHRASE` or `PULUMI_CONFIG_PASSPHRASE_FILE` when trying to access the Passphrase Secrets Provider; please ensure one of these environment variables is set to allow the operation to continue
```

If the variable is defined I get the same behavior:
`Migrating old configuration and state to new secrets provider`, 

with no prompt to set a new passphrase

**Comment by aszenz** [comment]
It looks like when the `PULUMI_CONFIG_PASSPHRASE` is set , the command `pulumi stack change-secrets-provider passphrase` actually sets it as the new passphrase without even asking for the old passphrase, to me it seems like a security issue but i'm not sure what's going on.

**Comment by idsvandermolen** [comment]
Seems to work fine in Pulumi `v3.9.1`. With `PULUMI_CONFIG_PASSPHRASE` env var set, the `pulumi stack change-secrets-provider passphrase` command correctly changes the password:
```bash
pulumi stack change-secrets-provider passphrase
Enter your new passphrase to protect config/secrets: 
Re-enter your new passphrase to confirm: 
Migrating old configuration and state to new secrets provider
```

Although it did take me quite a bit of time this was the command you need to run to change the secret. Bit of a hack, you would expect a command like `pulumi stack change-passphrase` or something.

**Comment by lukehoban** [comment]
> Due to 6286

> Seems to work fine in Pulumi v3.9.1.

I've confirmed that this was fixed by https://github.com/pulumi/pulumi/pull/6893 (which resolved #6286).
